rflx/model/session.py

```diff
@@ -82,10 +92,28 @@ def identifier(self) -> ID:
     def transitions(self) -> Sequence[Transition]:
         return self.__transitions or []
 
+    @property
+    def exception_transition(self) -> Optional[Transition]:
+        return self.__exception_transition
+
     @property
     def actions(self) -> Sequence[stmt.Statement]:
         return self.__actions
 
+    @property
+    def has_exceptions(self) -> bool:
+        return any(
+            isinstance(a, (stmt.Append, stmt.Extend))
+            or (
+                isinstance(a, stmt.Assignment)
+                and isinstance(
+                    a.expression,
+                    (expr.Selected, expr.Head, expr.Comprehension, expr.MessageAggregate),
+                )
```

Comment:
Is that sufficient? What if we had an assignment of the form `Result := Some_List'Head > Some_Other_List'Head`? I guess we need to evaluate this recursively.

Reply:
I see.